expression/builtin_control.go

```diff
@@ -164,7 +164,7 @@ func (c *caseWhenFunctionClass) getFunction(ctx sessionctx.Context, args []Expre
 	}
 	fieldTp.Decimal, fieldTp.Flen = decimal, flen
 	if fieldTp.EvalType().IsStringKind() && !isBinaryStr {
-		fieldTp.Charset, fieldTp.Collate = mysql.DefaultCharset, mysql.DefaultCollationName
+		fieldTp.Charset, fieldTp.Collate = charset.CharsetUTF8MB4, charset.CollationUTF8MB4
```

Comment:
/cc @XuHuaiyu , use `mysql.DefaultCharset` or `charset.CharsetUTF8MB4`?

Reply:
@XuHuaiyu, `DefaultCharset` is better. It hides the implementation details, once we change the default charset again, the code modification can be minimized.

Reply:
no, this place should be utf8mb4, if we did not set the charset, it can be default charset, but in here, it should definitely be utf8mb4.